qiskit/transpiler/preset_passmanagers/common.py

```diff
@@ -217,7 +217,7 @@ def generate_unroll_3q(
             target=target,
         )
     )
-    unroll_3q.append(HighLevelSynthesis(hls_config=hls_config))
+    unroll_3q.append(HighLevelSynthesis(hls_config=hls_config, coupling_map=None, target=target))
```

Comment:
I'm wondering if there is a potential issue here because unroll3q is typically run before layout so the qubit indices that get set in the pass will not be indices in the coupling map (ie qubit 0 in the circuit is not yet decided to be qubit 0 on the target backend). Maybe we need a flag to skip the qubits or otherwise flag that we've yet to set a layout.

Reply:
That's a good point. At the moment I think having `qubits=None` is exactly an indication that the layout has not yet been chosen. I have added a sentence on that to the plugin reference docs.

Reply:
But won't this still set `qubits` to a value because it's unconditionally populated in the `plugin_method.run()` call?

Reply:
Update: I spoke with Matthew in private, and he has convinced me that `HighLevelSynthesis` needs to know whether it's running before or after the layout is set, and suggested to add a flag `use_qubit_indices` to `HighLevelSynthesis` transpiler pass. This is now done in https://github.com/Qiskit/qiskit-terra/pull/10477/commits/e3211fbe69dcd93c471ca1de09157b965ccc3cbf.